.gitignore

```diff
@@ -22,6 +22,9 @@ var/
 *.egg-info/
 .installed.cfg
 *.egg
+requirements_for_test_common.txt
+pyproject.toml
```

Comment:
hmmm i'm not sure about this - pyproject.toml is a replacement for setup.py, and it was on my todo list to move all the project metadata from setup.py to pyproject.toml. Not having the file in the repo source would probably get in the way of it being parsed. Note that setup.py is not deprecated - https://packaging.python.org/en/latest/discussions/setup-py-deprecated/#setup-py-deprecated - but the movement in the python ecosystem is away from it.

i think ultimately the best solution may be to do a bit more work and write a script (using tomllib https://tomlkit.readthedocs.io/en/latest/) to merge in the `[tool.*]` sections from the utils template into the rest of the data 🤔

Reply:
What if we keep `pyproject.toml` in version control for now, but still copy it across from utils? I’d support doing the work to parse and merge later, if we decide to move away from `setup.py`

Reply:
yeah sounds good, lets do that for now 👍

might be nice to pop a comment at the top of the utils pyproject.toml saying "this file was generated based on notification_utils" or similar

Reply:
My script does that automatically 😄 :

<img width="693" alt="image" src="https://github.com/user-attachments/assets/a7c403f2-2c80-4225-8006-15a63b1b3208">

***

I’ve kept `pyproject.toml` in source control now and squashed that change into 72de27e

Reply:
🧙 🪄